---
int32 res
int32[] errorID